openedx/features/edly/management/commands/transalate_with_google.py

```diff
@@ -0,0 +1,152 @@
+
+
+def call_google_translation(text, target_language):
+    """
+    Call google translation api with help of google cloud library.
+    Parameters:
+        text (str): The text to be translated.
+        target_language (str): The target language code (e.g., 'id', 'ar', etc.).
+
+    Returns:
+        str: The translated text.
+    """
+    try:
+        result = client_instance.translate(
+            text,
+            target_language
+        )
+        return result["translatedText"]
+
+    except ValueError:
+        logging.error("API Error for msgid: %s", text)
+
+
+def translate_with_google(po_file, target_language):
+    """
+    Translate the .po file using the Google Translation API and update the .po entries.
+
+    Parameters:
+        po_file (str): The path to the .po file to be translated.
+        target_language (str): The target language code (e.g., 'id', 'ar', etc.).
+    """
+    pofile = polib.pofile(po_file, encoding='utf-8')
+
+    for entry in pofile:
+        try:
+            if not entry.msgstr:
+                has_plural = entry.msgid_plural is not None and entry.msgid_plural != ''
+                if not has_plural:
+                    entry.msgstr = call_google_translation(entry.msgid, target_language)
+
+                else:
+                    if 0 not in entry.msgstr_plural.keys():
+                        entry.msgstr_plural[0] = call_google_translation(entry.msgid, target_language)
+                    if 1 not in entry.msgstr_plural.keys():
+                        entry.msgstr_plural[1] = call_google_translation(entry.msgid_plural, target_language)
+
+                entry.comment = COMMENT
+                pofile.save()
+
+        except (ValueError, IOError) as error:
+            logging.error("Error '%s' while writing to .po file for msgid: '%s'.", error, entry.msgid)
+
+    logging.info("Translation completed for %s.", target_language)
+
+
+def po_file_to_csv(csv_file_path, po_file_path):
+    """
+    Export translations from the .po file to a CSV file.
+
+    Parameters:
+        csv_file_path (str): The path to the CSV file.
+        po_file_path (str): The path to the .po file.
+    """
+
+    with open(csv_file_path, 'a', newline='', encoding='utf-8') as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerow(['String', 'Translated String', 'Source'])
+
+    pofile = polib.pofile(po_file_path, encoding='utf-8')
+    for entry in pofile:
+        has_plural = entry.msgid_plural is not None and entry.msgid_plural != ''
+
+        if entry.comment == COMMENT:
+            source = "Google Translation"
+        else:
+            source = "Transifex"
+
+        if not has_plural:
+            write_to_csv(csv_file_path, entry.msgid, entry.msgstr, source)
+        else:
+            write_to_csv(csv_file_path, entry.msgid, entry.msgstr_plural[0], source)
+            write_to_csv(csv_file_path, entry.msgid_plural, entry.msgstr_plural[1], source)
+
+
+if __name__ == "__main__":
```

Comment:
@mfarhan943 we can inherit `BaseCommand` class and implement it like this 
https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/#module-django.core.management

Reply:
Creating a Django management command is currently challenging because of Google Cloud's incompatibility with the current edx-platform Python version. Additionally, Google authentication requires a JSON credential file. My suggestion is to continue with the current approach, but I can convert it to a management command if that is preferred.